src/com/oltpbenchmark/api/SchemaManager.java

```diff
@@ -0,0 +1,298 @@
+            .column("c_zip", "char(9) NOT NULL")
+            .column("c_phone", "char(16) NOT NULL")
+            .column("c_since", "timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP")
+            .column("c_middle", "char(2) NOT NULL")
+            .column("c_data", "varchar(500) NOT NULL")
+            .primaryKey("((c_w_id,c_d_id) HASH,c_id)")
+        .build(),
+        Table.Builder()
+            .name(TPCCConstants.TABLENAME_DISTRICT)
+            .column("d_w_id", "int NOT NULL")
+            .column("d_id", "int NOT NULL")
+            .column("d_ytd", "decimal(12,2) NOT NULL")
+            .column("d_tax", "decimal(4,4) NOT NULL")
+            .column("d_next_o_id", "int NOT NULL")
+            .column("d_name", "varchar(10) NOT NULL")
+            .column("d_street_1", "varchar(20) NOT NULL")
+            .column("d_street_2", "varchar(20) NOT NULL")
+            .column("d_city", "varchar(20) NOT NULL")
+            .column("d_state", "char(2) NOT NULL")
+            .column("d_zip", "char(9) NOT NULL")
+            .primaryKey("((d_w_id,d_id) HASH)")
+        .build(),
+        Table.Builder()
+            .name(TPCCConstants.TABLENAME_ITEM)
+            .column("i_id", "int NOT NULL")
+            .column("i_name", "varchar(24) NOT NULL")
+            .column("i_price", "decimal(5,2) NOT NULL")
+            .column("i_data", "varchar(50) NOT NULL")
+            .column("i_im_id", "int NOT NULL")
+            .primaryKey("(i_id)")
+        .build(),
+        Table.Builder()
+            .name(TPCCConstants.TABLENAME_WAREHOUSE)
+            .column("w_id", "int NOT NULL")
+            .column("w_ytd", "decimal(12,2) NOT NULL")
+            .column("w_tax", "decimal(4,4) NOT NULL")
+            .column("w_name", "varchar(10) NOT NULL")
+            .column("w_street_1", "varchar(20) NOT NULL")
+            .column("w_street_2", "varchar(20) NOT NULL")
+            .column("w_city", "varchar(20) NOT NULL")
+            .column("w_state", "char(2) NOT NULL")
+            .column("w_zip", "char(9) NOT NULL")
+            .primaryKey("(w_id)")
+        .build()
+    ).collect(Collectors.toMap(Table::name, e -> e));
+
+    SchemaManager(Connection db_connection) {
+        this.db_connection = db_connection;
+    }
+
+    public static Set<String> getTableNames() {
+        return tables.keySet();
+    }
+
+    public static String getInsertDml(String table_name) {
+        Table t = tables.get(table_name);
+        if (t == null) {
+            throw new RuntimeException(String.format("Unknown table: %s", table_name));
+        }
+        return t.getInsertDml();
+    }
+
+    private void execute(String sql) throws SQLException {
+        LOG.info(sql);
+        db_connection.createStatement().execute(sql);
+    }
+
+    void create() throws SQLException {
+        for (Table t : tables.values()) {
+            execute(t.getDropDdl());
+        }
+        for (Table t : tables.values()) {
+            execute(t.getCreateDdl());
```

Comment:
I believe execute("create table ....."); would have been a bit simpler.
Not sure if we would ever need the columns/table classes other than for creating the tables on the database.

Reply:
the code being replaces was using hsqldb to introspect the schema of an in memory replica of the database to create insert statements with the appropriate number of wildcards. by separating column defs out, we can still support this